src/main/java/io/cdap/plugin/dlp/SensitiveRecordRedaction.java

```diff
@@ -288,19 +300,40 @@ private DlpServiceSettings getSettings() throws IOException {
     @Macro
     @Nullable
     private String templateId;
-    private static final Gson gson = new GsonBuilder()
+
+    private static final Gson GSON = new GsonBuilder()
       .registerTypeAdapter(DlpFieldTransformationConfig.class, new DlpFieldTransformationConfigCodec())
       .create();
 
     public List<DlpFieldTransformationConfig> parseTransformations() throws Exception {
-      String[] values = gson.fromJson(fieldsToTransform, String[].class);
+      String[] values = GSON.fromJson(fieldsToTransform, String[].class);
       List<DlpFieldTransformationConfig> transformationConfigs = new ArrayList<>();
       for (String value : values) {
-        transformationConfigs.add(gson.fromJson(value, DlpFieldTransformationConfig.class));
+        transformationConfigs.add(GSON.fromJson(value, DlpFieldTransformationConfig.class));
       }
       return transformationConfigs;
     }
 
+    /**
+     * Get the set of fields that are being transformed or are required for transforms to work. This is used to limit
+     * the payload size to DLP endpoints, the transform will only send the values of the required fields.
+     *
+     * @return Set of field names
+     */
+    public Set<String> getRequiredFields() {
+
+
+      try {
+        HashSet<String> results = new HashSet<>();
+        parseTransformations().stream()
+          .map(DlpFieldTransformationConfig::getRequiredFields)
+          .forEach(results::addAll);
+        return results;
+      } catch (Exception e) {
+        LOG.error(String.format("Failed to get list of required fields: %s", e.getMessage()));
```

Comment:
If it is not an error, don't log it with ERROR level. Also, when logging an exception, always include the exception in the log. Also, don't use String.format in log (use `{}` placeholder instead).

```
LOG.warn("Unable to get list of required fields, defaulting to an empty set", e);
```

Reply:
done